### ISSUE TITLE
"gulp trust-dev-cert" command not working for SharePoint Framework project creation 

### ISSUE THREAD

**Issue by Kamlesh780** [SharePoint/sp-dev-docs#6703]
Thank you for reporting an issue or suggesting an enhancement. We appreciate your feedback - to help the team to understand your needs, please complete the below template to ensure we have the necessary details to assist you.

**Category**

- [ ]  Question
- [x]  Typo
      **-  Bug -- It is a bug**
- [ ]  Additional article idea

**Expected or Desired Behavior**

I have created my first HelloWorld SPFx project. But before running my first project I need to run 'gulp trust-dev-cert not' command. 
My solution **HelloWorld** is successfully created only the problem is that the **gulp trust-dev-cert** command is not working.
I'm trying to execute this command in the project folder and getting the below error please help.

**Environment details (*development & target environment*)**
- **Developer Environment**: Windows 10
- **Target Environment**: SharePoint Online (Local Workbench)
- **Framework**: Node.js v10
- **Browser(s)**: Chrome v79 | Edge 
- **Tooling**: VS Code | @microsoft/generator-sharepoint@1.11.0
- **Additional details**: [*Node Version*: v10.23.3], [*Gulp Version*: CLI version: 2.3.0 | Local version: 3.9.1]

**Observed Behavior**

C:\Users\Kamlesh\HelloWorld>yo @microsoft/sharepoint

     _-----_
    |       |    .--------------------------.
    |--(o)--|    |      Welcome to the      |
   `---------´   |  SharePoint Client-side  |
    ( _´U`_ )    |    Solution Generator    |
    /___A___\    '--------------------------'
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

Let's create a new SharePoint solution.
? What is your solution name? hello-world
? Which baseline packages do you want to target for your component(s)? SharePoint Online only (latest)
? Where do you want to place the files? Use the current folder
Found npm version 6.14.11
? Do you want to allow the tenant admin the choice of being able to deploy the solution to all sites immediately without running any feature deployment or adding apps i
n sites? No
? Will the components in the solution require permissions to access web APIs that are unique and not shared with other components in the tenant? No
? Which type of client-side component to create? WebPart
Add new Web part to solution hello-world.
? What is your Web part name? HelloWorld
? What is your Web part description? HelloWorld description
? Which framework would you like to use? React

   create package.json
   create config\package-solution.json
   create config\config.json
   create config\serve.json
   create tsconfig.json
   create .vscode\extensions.json
   create .vscode\launch.json
   create .vscode\settings.json
   create config\copy-assets.json
   create config\deploy-azure-storage.json
   create config\write-manifests.json
   create src\index.ts
   create gulpfile.js
   create README.md
   create tslint.json
   create .editorconfig
   create .gitignore
   create src\webparts\helloWorld\components\IHelloWorldProps.ts
   create src\webparts\helloWorld\components\HelloWorld.module.scss
   create src\webparts\helloWorld\components\HelloWorld.tsx
   create src\webparts\helloWorld\HelloWorldWebPart.ts
   create src\webparts\helloWorld\loc\en-us.js
   create src\webparts\helloWorld\loc\mystrings.d.ts
   create src\webparts\helloWorld\HelloWorldWebPart.manifest.json
   create teams\9626ec87-8866-4bd8-a877-9b713d8687d3_outline.png
   create teams\9626ec87-8866-4bd8-a877-9b713d8687d3_color.png

npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated natives@1.1.6: This module relies on Node.js's internals and will break at some point. Do not use it, and update to graceful-fs@4.x.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated es6-collections@0.5.6: not actively maintained anymore
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated core-js@1.2.7: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated mkdirp-promise@5.0.1: This package is broken and no longer maintained. 'mkdirp' itself supports promises now, please switch to that.
npm WARN deprecated kleur@2.0.2: Please upgrade to kleur@3 or migrate to 'ansi-colors' if you prefer the old syntax. Visit <https://github.com/lukeed/kleur/releases/tag/v3.0.0\> for migration path(s).
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.

> deasync@0.1.21 install C:\Users\Kamlesh\HelloWorld\node_modules\deasync
> node ./build.js

`win32-x64-node-10` exists; testing
Binary is fine; exiting

> node-sass@4.12.0 install C:\Users\Kamlesh\HelloWorld\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.12.0/win32-x64-64_binding.node
Download complete..] - :
Binary saved to C:\Users\Kamlesh\HelloWorld\node_modules\node-sass\vendor\win32-x64-64\binding.node
Caching binary to C:\Users\Kamlesh\AppData\Roaming\npm-cache\node-sass\4.12.0\win32-x64-64_binding.node

> core-js@2.6.12 postinstall C:\Users\Kamlesh\HelloWorld\node_modules\core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)

> uglifyjs-webpack-plugin@0.4.6 postinstall C:\Users\Kamlesh\HelloWorld\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

> node-sass@4.12.0 postinstall C:\Users\Kamlesh\HelloWorld\node_modules\node-sass
> node scripts/build.js

Binary found at C:\Users\Kamlesh\HelloWorld\node_modules\node-sass\vendor\win32-x64-64\binding.node
Testing binary
Binary is fine
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules\watchpack\node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN ajv-keywords@3.5.2 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.
npm WARN uglifyjs-webpack-plugin@0.4.6 requires a peer of webpack@^1.9 || ^2 || ^2.1.0-beta || ^2.2.0-rc || ^3.0.0 but none is installed. You must install peer dependencies yourself.

added 1797 packages from 1058 contributors and audited 1807 packages in 3904.47s

37 packages are looking for funding
  run `npm fund` for details

found 2043 vulnerabilities (1975 low, 13 moderate, 55 high)
  run `npm audit fix` to fix them, or `npm audit` for details

      _=+#####!
   ###########|       .------------------------------------.
   ###/    (##|(@)    |          Congratulations!          |
   ###  ######|   \   |  Solution hello-world is created.  |
   ###/   /###|   (@) |   Run gulp serve to play with it!  |
   #######  ##|   /   '------------------------------------'
   ###     /##|(@)
   ###########|
      **=+####!

**C:\Users\Kamlesh\HelloWorld>gulp trust-dev-cert
Build target: DEBUG
[15:49:07] Using gulpfile ~\HelloWorld\gulpfile.js
[15:49:07] Starting gulp
[15:49:07] Starting 'trust-dev-cert'...
[15:49:07] Starting subtask 'configure-sp-build-rig'...
[15:49:07] Finished subtask 'configure-sp-build-rig' after 55 ms
[15:49:07] Starting subtask 'trust-cert'...
[15:49:13] Error - [trust-cert] TypeError: Cannot read property 'toString' of null
[15:49:13] Error - 'trust-cert' sub task errored after 6.12 s
 Cannot read property 'toString' of null
[15:49:13] 'trust-dev-cert' errored after 6.22 s
[15:49:13]
[15:49:14] ==================[ Finished ]==================
Error - [trust-cert] TypeError: Cannot read property 'toString' of null
Error - 'trust-cert' sub task errored after 6.12 s
 Cannot read property 'toString' of null
[15:49:14] Project hello-world version:0.0.1
[15:49:14] Build tools version:3.12.1
[15:49:14] Node version:v10.23.3
[15:49:14] Total duration:4.25 min
[15:49:14] Task errors:2**

C:\Users\Kamlesh\HelloWorld>

Thanks for your contribution! Sharing is caring.

**Comment by andrewconnell** [comment]
Respectfully, I've repeatedly asked you to please use the issue template for this repo. It contains instructions that **we need you to follow in order to help**. Specifically, no-where have you said what version of SPFx you're using, what your environment is, etc.... this information is requested because it's needed.

Specifically, you deleted the following from the new issue template. Please complete this as it contains NECESSARY details:

## Environment details (*development & target environment*)

> **DELETE THIS LINE BEFORE SUBMITTING** - Complete the following (*if ignored you'll be prompted for it before we can address your issue... save the time and provide it in your initial submission*).

- **Your Developer Environment**: [e.g. Windows 10 | MacOS 10.15.x]
- **Target Environment**: [e.g. SharePoint Server 2016 | SharePoint Server 2019 | SharePoint Online]
- **Framework**: [e.g. .NET Framework v3.x | .NET Core v3.x | Node.js v6/v8/v10/v12]
- **Browser(s)**: [e.g. Chrome v79 | Safari v22 | FireFox v31 | IE v11 | Edge | Chromium Edge v79]
- **Tooling**: [e.g. VS Code | SPFx v1.10.0 | Visual Studio 2019]
- **Additional details**: *The more context you can provide, the easier it is (and therefore quicker) to help.*

**Comment by Kamlesh780** [comment]
Hi @andrewconnell,
I have updated my query. Can you please check it?
Let me know if you need any additional details.

Thank you :)

**Comment by andrewconnell** [comment]
@Kamlesh780 

> Let me know if you need any additional details.

What version of the SharePoint Framework you're using. That's in the **Tooling** request.

**Comment by Kamlesh780** [comment]
> @Kamlesh780
> 
> > Let me know if you need any additional details.
> 
> What version of the SharePoint Framework you're using. That's in the **Tooling** request.

C:\Users\Kamlesh>npm ls -g --depth=0
C:\Users\Kamlesh\AppData\Roaming\npm
+-- @microsoft/generator-sharepoint@1.11.0
+-- gulp@4.0.2
`-- yo@3.1.1

**Comment by andrewconnell** [comment]
What you have installed looks correct... if you upgraded the version of Node at any point after upgrading packages, that could have some impact. All google searches of this error indicate it comes from Node.js v8... so not sure where to go from here.

When you have all prereq's installed and the correct versions, but still have errors, I suggest you uninstall all global packages, your Node.js install, and then re-install clean (Node.js v10 + yeoman, gulp & the SPFx generator).

**Comment by Kamlesh780** [comment]
> What you have installed looks correct... if you upgraded the version of Node at any point after upgrading packages, that could have some impact. All google searches of this error indicate it comes from Node.js v8... so not sure where to go from here.
> 
> When you have all prereq's installed and the correct versions, but still have errors, I suggest you uninstall all global packages, your Node.js install, and then re-install clean (Node.js v10 + yeoman, gulp & the SPFx generator).

I tried 2 times. But it is still not working.

**Comment by Tobor58** [comment]
This is question
I have successfully created a hello world once - gulp serve  / gulp trust-dev-cert / success saw the local workbench etc. no issue.
  
I created a 2nd new web part in different director attempted to serve the new part and could NOT see the part on local workbench etc. 
attempts to run   gulp trust-dev-cert for the NEW part complete but no new cert is seems to be created I am not prompted to accept the new cert etc. no matter what happens the new part cannot be viewed on the machine I developed it on using the local workbench. 
The gist of this is:
The previously developed part works  on the local bench - the new one won't.
I can use the workbench of my SharePoint site (on Prem) to view the NEW part no issue - so I know the part works. 

Do  I need to delete the previously created cert - then create a NEW cert for the new project? 
How can I correctly identify the previously created DEV cert and remove it correctly if I need to?
Thank you.

**Comment by andrewconnell** [comment]
@Tobor58 Do the projects use different versions of SPFx? Different versions of Node? If so, the cert may have changed & thus, you'd need to trust the one that's giving you an issue.

BTW, you don't need to remove the old one.

**Comment by Tobor58** [comment]
@andrewconnell  great to "meet you" and thank you.

I neglected to say that :  I never saw the dialog box to trust the newly created part. and there was no error  
so perhaps the new cert is not getting created for the new part? 

I have only created two parts on the machine I was using.

example of the directory structure 

D:\---\>MyParts>
       \---- >Part_1  (worked,  showed dialog, trusted. ) 
       \---- >Part_2  (no dialog shown after gulp trust-dev-cert)

**Comment by andrewconnell** [comment]
@Kamlesh780 & @Tobor58 

This isn't a bug... sometimes things get a little screwy when trusting dev certs. There are numerous closed issues on this topic that you can research. But when things get messed up, easiest to just clean & start over:

1) untrust the cert: **gulp untrust-dev-cert**
2) delete old certs potentially left around (ie: delete the `~/.gcb-serve-data` folder... [google for references](https://www.google.com/search?q=gcb-serve-data))
3) trust the cert in a project created with the most recent version of the SPFx generator installed... to be SURE it works, do this as admin depending on your platform (Windows = admin command prompt / macOS = sudo): **gulp trust-dev-cert**

**Comment by Kamlesh780** [comment]
> @Kamlesh780 & @Tobor58
> 
> This isn't a bug... sometimes things get a little screwy when trusting dev certs. There are numerous closed issues on this topic that you can research. But when things get messed up, easiest to just clean & start over:
> 
> 1. untrust the cert: **gulp untrust-dev-cert**
> 2. delete old certs potentially left around (ie: delete the `~/.gcb-serve-data` folder... [google for references](https://www.google.com/search?q=gcb-serve-data))
> 3. trust the cert in a project created with the most recent version of the SPFx generator installed... to be SURE it works, do this as admin depending on your platform (Windows = admin command prompt / macOS = sudo): **gulp trust-dev-cert**

It's working for me. Grateful for your support. 
Thank you :)
